src/components/popover-menu/popover-menu-button.tsx

```diff
@@ -0,0 +1,21 @@
+import { PopoverButton as HeadlessUiPopoverButton } from "@headlessui/react";
+import React from "react";
+import { usePopoverMenuContext } from "./popover-menu-context";
+import { Button, ButtonProps } from "../button/button";
+
+export interface NavigationPopoverButtonProps extends ButtonProps {
+    children: React.ReactNode;
+    onClick?: () => void;
+}
+
+export const PopoverMenuButton = ({ onClick, ...restProps }: NavigationPopoverButtonProps) => {
```

Comment:
Please render the children element explicitly, instead of passing it inside ...restProps. AFAIK passing children as a prop is an antipattern

Reply:
check check ✅